libdrgn/dwarf_info.c

```diff
@@ -6816,6 +6816,45 @@ drgn_type_from_dwarf_internal(struct drgn_debug_info *dbinfo,
 	return NULL;
 }
 
+static struct drgn_error *
+find_enclosing_namespace(struct drgn_namespace_dwarf_index *global_namespace,
+			 const char **name, size_t *name_len,
+			 struct drgn_namespace_dwarf_index **namespace_ret)
+{
+	*namespace_ret = global_namespace;
+	if (*name_len >= 2 && memcmp(*name, "::", 2) == 0) {
+		/* Explicit global namespace. */
+		*name_len -= 2;
+		*name += 2;
+	}
+
+	const char *template_parameters_start = memchr(*name, '<', *name_len);
+	ptrdiff_t searchable_len =
+		template_parameters_start ?
+				  ((uintptr_t)template_parameters_start) - ((uintptr_t)*name) :
+				  *name_len;
```

Comment:
Why are the casts to `uintptr_t` necessary here? Can't this be:

```suggestion
				  template_parameters_start - *name:
```